src/test/java/spoon/test/method/MethodTest.java

```diff
@@ -85,4 +86,18 @@ public void testGetAllMethods() throws Exception {
 		assertEquals(1, methods.stream().filter(method -> "foo".equals(method.getSimpleName())).count());
 	}
 
+	@Test
+	public void testGetAllMethodsAdaptingType() throws Exception {
+		// contract: AbstractTypingContext should not enter in recursive calls when resolving autoreferenced bounding type
+		// such as T extends Comparable<? super T>
+		Launcher l = new Launcher();
+		l.getEnvironment().setNoClasspath(true);
+		l.addInputResource("src/test/resources/noclasspath/spring/PropertyComparator.java");
+		l.buildModel();
+
+		CtType<?> propertyComparator = l.getModel().getElements(new NamedElementFilter<CtType>(CtType.class, "PropertyComparator")).get(0);
+		CtMethod method = propertyComparator.getMethodsByName("compare").get(0);
```

Comment:
Which call failed on StackOverflowError? 
A) getMethodsByName
B) getSignature

Reply:
getMethodsByName

Reply:
So why did you removed this method call from the test??

Reply:
:D sorry I answer yesterday evening and actually I was wrong: it's getAllMethods which led to the bug. My test wasn't failing after all yesterday.